### ISSUE TITLE
Get admin rights

### ISSUE THREAD

**Issue by JPFrancoia** [Digital-Sapphire/PyUpdater#35]
Hi,

# TL;DR

Can PyUpdater get admin rights to write in admin space ?

# Long version

We discussed that a lot for Esky, but the process was kind of broken: https://github.com/cloudmatrix/esky/issues/114

Let's suppose I have a wonderful software I want to ship to my Windows users. I will create a ```setup.exe``` with Inno setup (for example). The user will then just have to run the installer.

During the install, the files will be written in ```Program Files```. In order to do that, the user will give admin rights to the install process (on Windows, a popup will appear and the user will just have to click on a button to agree). Finally, the user will run the program peacefully.

Now let's say I release a new version of my program. It will try to update itself through PyUpdater's API: it will try to download the new version, but the update process will fail at this step, because the program does not have sufficient permissions to write in ```Program Files```.

The same problem exists on Mac OS. My workaround is to trick the install process into installing the program in user space (not in ```Program Files```), so it never has to ask for admin rights. But I would prefer not to do that.

Is there a solution with PyUpdater ?

It would require some kind of module to ask the user his password.

**Comment by JMSwag** [comment]
@JPFrancoia Currently no. I'd like to get this implemented though.

**Comment by JMSwag** [comment]
This will not be implemented at this point in time. I don't see a huge win for the time that will need to be invested.

**Comment by Devligue** [comment]
@JMSwag I just run in to the same problem as @JPFrancoia. Most windows users sadly install they software by default in Program Files on C. Since PyInstaller doesn't add admin elevation to exe.manifest AND PyUpdater without the privileges will just fail in the background and the user won't even know, I think it is quite important problem. It is up to user then if her runs the program as admin and everything works fine or he runs it normally and it fails silently. Not every user is a programmer, and from their perspective it will be just a shitty software that instead of updating just quits...

**Comment by JMSwag** [comment]
@Devligue When deploying to windows I install to the local appdata directory. I'll gladly accept any PR's that implement this logic.

**Comment by flutefreak7** [comment]
Interestingly anaconda implements UAC elevation during conda updates to things that create start menu icons like ipython, jupyter notebook, etc. I always end up wishing I had just run the update in an elevated CMD prompt because you often have to accept the UAC elevation prompt multiple times.

**Comment by flutefreak7** [comment]
Here's one example of elevating UAC prompt to do stuff: https://github.com/ContinuumIO/menuinst/blob/master/menuinst/windows/win_elevate.py

**Comment by JMSwag** [comment]
@flutefreak7 This is being worked on by @cameronmaske in #116
